File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dic = {}

    # 참가자 이름 등장 횟수 기록
    for p in participant:
        dic[p] = dic.get(p, 0) + 1

    # 완주자 이름 등장 횟수 차감
    for c in completion:
        dic[c] -= 1

    # 값이 1 이상인 key가 완주 못한 사람
    for key, value in dic.items():
        if value > 0:
            return key

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const hashMap = new Map();

    for(let i = 0; i < participant.length; i++) {
        let part = participant[i], 
            comp = completion[i];

        hashMap.set(part, (hashMap.get(part) || 0) + 1);
        hashMap.set(comp, (hashMap.get(comp) || 0) - 1);
    }
    for(let [key, value] of hashMap) {
        if(value > 0) return key;
    }
    return 'nothing';
}